src/CommunityToolkit.Maui/PopupService.cs

```diff
@@ -45,13 +47,39 @@ public PopupService()
 		services.AddTransient(typeof(TPopupViewModel));
 	}
 
+	/// <inheritdoc cref="IPopupService.ClosePopup(object?)" />
+	public void ClosePopup(object? result = null)
+	{
+		var popupReference = popupDisplayStack.Peek();
+
+		if (popupReference.TryGetTarget(out var popup))
+		{
+			popup.Close(result);
+		}
+	}
+
+	/// <inheritdoc cref="IPopupService.ClosePopupAsync(object?)" />
+	public Task ClosePopupAsync(object? result = null)
+	{
+		var popupReference = popupDisplayStack.Peek();
```

Comment:
this will return the same instance every time, the `Peek` method doesn't remove the item, use `Pop` instead.

Reply:
This doesn't pop, the closed handler currently pops the stack. This was to handle the scenario where a caller closes a popup without using our new Close method